### PR TITLE
Allow -nopervasives when running a toplevel

### PR DESCRIPTION
**Pull request by trefis**
Follow up to #1969

### CODE REVIEW

**Comment by trefis** [comment]
The initial title was a misnomer, to be able to pass `-nopervasives` in the testsuite I had to make the toplevels accept the flag.
I don't think there is any strong reason not to allow it, so I did that change in the first commit.

The second one actually passes the flag for some of the tests pointed out by @Octachron .

I'll add a Changes entry when once someone reviews.

**Review by gasche — APPROVED** [review]
This looks good, thanks! You may want to find the number of the main_args-revamp PR and add a cross-reference so that the author knows that a rebase is in order.

**Comment by trefis** [comment]
Ah, I thought that had been merged already.
Will do.

File: testsuite/tests/shadow_include/shadow_all.ml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 (* TEST
    * expect
+   flags = "-nopervasives" (* can't pass -nostdlib because of objects. *)
```
Comment (Octachron) [inline testsuite/tests/shadow_include/shadow_all.ml:3]: I was a bit worried about this part, but this file does not seem to cause issues in the various stdlib improvement PRs.

**Review by Octachron — APPROVED** [review]
It looks good to me too.

File: testsuite/tests/shadow_include/shadow_all.ml (hunk 1 above)
Comment (trefis) [reply testsuite/tests/shadow_include/shadow_all.ml:3]: Yes, I think the idents will shift if someone ever changes the content of `CamlinternalObj`. But I think I can live with that.

**Pull request merged**